fl4health/processing/VAE_dim_reduction.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC
+from pathlib import Path
+from typing import Tuple
+
+import numpy as np
+import torch
+
+from fl4health.model_bases.autoencoders_base import ConditionalVAE, VarioationalAE
+
+
+class Processing(ABC):
+    """
+    Abstact class to add processing methods.
+    """
+
+    def __init__(
+        self,
+        checkpointing_path: Path,
+    ) -> None:
+        self.checkpointing_path = checkpointing_path
+        self.DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+    def load_VAE(self) -> VarioationalAE:
+        autoencoder = torch.load(self.checkpointing_path)
+        autoencoder.eval()
+        return autoencoder
+
+    def load_CVAE(self) -> ConditionalVAE:
+        autoencoder = torch.load(self.checkpointing_path)
+        autoencoder.eval()
+        return autoencoder
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}"
+
+
+class VAEProcessor(Processing):
+    """Transformer processor to encode the data using VAE encoder."""
+
+    def __init__(
+        self,
+        checkpointing_path: Path,
+    ) -> None:
+        super().__init__(checkpointing_path)
+        self.autoencoder: VarioationalAE = self.load_VAE()
+
+    def __call__(self, sample: torch.Tensor) -> torch.Tensor:
+        # This transformer is called for the input samples after they are transfered into toch tensors.
+        mu, logvar = self.autoencoder.encode(sample)
+        return mu.clone().detach()
+
+
+class LabelConditionedProcessor(Processing):
+    """Transformer processor to encode the data using CVAE encoder, with the labels as conditions."""
```

Comment:
Just curious because I think I must be missing something. When would we be able to condition on the labels? Since in most cases I assume we won't have access to the ground truth labels.

Reply:
It's a very good point! I should probably just keep the LabelConditionedVAE for conditional synthetic data generation.
I initially thought it could also be used as some sort of sample-specific condition (which could be anything, not necessarily the target), but that doesn't make sense for dimensionality reduction now that I am thinking more about it.

Reply:
Ah I see, it could potentially come in handy in the future! If we need to add something like that in the future we can refer back to this PR :)